### DISCUSSION TITLE
How can I silence the ST console outputting {{user}} prompts?

### DISCUSSION THREAD

**Discussion by hugalafutro** [SillyTavern/SillyTavern#3171] [Q&A] [answered]
Question in title, not much to elaborate. Would prefer my coomer chats not logged anywhere if at all possible ;) I use docker if that matters, I tried researching silencing the whole container output, but something seemingly so simple seems to be impossible to achieve as well.

**Comment by stanrust** [comment]
console.log = () => {}

**Reply by hugalafutro** [reply]
Could you elaborate a bit? I tried putting the line to config file but no bueno. Where do I put it?

**Reply by stanrust** [reply]
ST uses **console.log** for logging. You can suppress these logs by redefining the function to do nothing.
Edit **server.js** and add the following line to the top: **console.log = () => {};**

**Reply by Cohee1207** [reply]
Please DON'T patch code files directly, it's an extremely bad practice since it will cause merge conflicts on git pull which increases support burden. Better make a server plugin like this (just download and drop to /plugins):
https://gist.github.com/Cohee1207/712a4e604350876f7fe3591ac37d9ff0

Requires a config.yaml flag to be enabled. Check the documentation: https://docs.sillytavern.app/for-contributors/server-plugins/

**Reply by Cohee1207** [reply]
Docker may require creating an additional volume for plugins if you haven't already. Also covered in the docs: https://docs.sillytavern.app/installation/docker/#running-server-plugins

**Reply by hugalafutro** [reply]
@Cohee1207 thank you that is a very elegant solution for docker! I was gonna post that is going to be a bother to edit the file each time the image updates, but this solves it. I already have the plugins dir mounted from when I was messing with some extensions.

**Reply by hugalafutro** [reply]
works as advertised, thanks again!
absolutely nothing is printed in the console after this output.

![image](https://github.com/user-attachments/assets/cd49124a-b3cc-447a-b6f8-b4030afdd32b)

**Reply by hugalafutro** [reply, marked as answer]
Just a quick note to anyone from the future:

This method  DOES NOT work and IS NOT NEEDED anymore in 1.12.12
from changelog:
- Server: Added console logging level configuration.


Now it's configured in `config.yaml` here: https://github.com/SillyTavern/SillyTavern/pull/3307/files#diff-08051af2e0078867a8be0daf177bd361b0fd8ab6bd0bca3b0d4231c14fdf2ca8

from: https://github.com/SillyTavern/SillyTavern/pull/3307
> Added an option in the config.yaml file to select the minimum log level to display in the terminal. Removing DEBUG messages will hide the prompts.
```
# Minimum log level to display in the terminal (DEBUG = 0, INFO = 1, WARN = 2, ERROR = 3)
minLogLevel: 0
```

**Reply by Cohee1207** [reply]
^^ This should be the answer instead.

**Comment by Cohee1207** [comment]
Look here: https://github.com/SillyTavern/SillyTavern/issues/2720#issuecomment-2322767556

**Reply by hugalafutro** [reply]
I should have probably mentioned I'm on linux, running in docker container so can't use the vbscript method.

I originally thought it would be a setting in config.yaml, when I found none, I researched ways to silence log output from whole container, but that proved seemingly impossible as well.

Like I don't mind seeing the initialization info, ip address etc, just don't want to see chat in console
